packages/interim/src/renovation/component_wrapper/common/mutations_recording.ts

```diff
@@ -1,27 +1,26 @@
 function isChildNode(node: Node): node is ChildNode {
-    return typeof (node as Partial<ChildNode>).remove === 'function';
+  return typeof (node as Partial<ChildNode>).remove === 'function';
+}
+
+function revertMutation({ type, addedNodes }: MutationRecord): void {
+  switch (type) {
+    case 'childList':
+      addedNodes.forEach((n) => isChildNode(n) && n.remove());
+      break;
+    default:
+      break;
   }
-  
-  function revertMutation({ type, addedNodes }: MutationRecord): void {
-    switch (type) {
-      case 'childList':
-        addedNodes.forEach((n) => isChildNode(n) && n.remove());
-        break;
-      default:
-        break;
-    }
-  }
-  
-  export function recordMutations<TReturn>(target: Node, func: () => TReturn): () => void {
-    const observer = new MutationObserver(() => {});
-    // eslint-disable-next-line spellcheck/spell-checker
-    observer.observe(target, { childList: true, subtree: false });
-  
-    func();
-  
-    const mutations = observer.takeRecords();
-    observer.disconnect();
-  
-    return () => mutations.forEach(revertMutation);
-  }
-  
\ No newline at end of file
+}
+
+export function recordMutations<TReturn>(target: Node, func: () => TReturn): () => void {
```

Comment:
Minor comment.

Maybe we should change the generic ```TReturn``` to the ```unknown``` type?
Something like this:
```suggestion
export function recordMutations(target: Node, func: () => unknown): () => void {
```

Because we lost the ```func`` return value and it doesn't matter what this function returns :)

Reply:
Yep, we should clean up it. Since this PR is for synchronisation I've included proposed changes to this one: https://github.com/DevExpress/DevExtreme/pull/23637